realm/src/main/java/io/realm/Realm.java

```diff
@@ -1198,6 +1202,20 @@ boolean contains(Class<? extends RealmObject> clazz) {
     }
 
     /**
+     * Prepare an async query
+     * TODO point to documentation or example about async query
+     * @param clazz The class of the object which is to be queried for
+     * @return A typed RealmQuery, which can be used to query for specific objects of this type
+     * @throws java.lang.RuntimeException Any other error
+     * @see io.realm.RealmQuery
+     * @param callback
+     */
+    public <E extends RealmObject> AsyncRealmQuery<E> asyncWhere(Class<E> clazz, Realm.AsyncCallback<RealmResults<E>> callback) {
```

Comment:
Any reason you moved from ```.findAllAsync(callback)``` to asyncWhere(callback). The former looks much more readable to me?